### DISCUSSION TITLE
Can you create a URI to a setting in the settings panel?

### DISCUSSION THREAD

**Discussion by JustinGrote** [microsoft/vscode-discussions#1716] [Extension Development QnA] [answered]
I want to pop up a dialog. ShowMessage supports limited markdown with links, so is there a way to craft a link so that when clicked it takes the user to a specific setting in the UI? vscode:// maybe? Do I have to create a custom command?

**Comment by lramos15** [comment]
You can try a command link and see if that works for you!

The format of the link would look like this
```
`command:workbench.action.openSettings?${encodeURIComponent('["SETTING_ID"]')}`
```

**Reply by JustinGrote** [reply]
I had no idea that was a thing, I'll give it a try, thanks!

**Reply by lramos15** [reply]
On your markdown string you will need to trust the command to execute. Something like
`messageString.isTrusted = { enabledCommands: [openSettingsCommand] };` or `messageString.isTrusted = true;` should do it for you

**Reply by JustinGrote** [reply]
Does this work via the LSP? Not sure how I'd set that via a ShowNotification

**Reply by lramos15** [reply]
Sorry I'm not as familiar with LSP. Maybe @dbaeumer can chime in as they have more familiarity with LSP

**Comment by dbaeumer** [comment, marked as answer]
@JustinGrote yes, but you need to opt into commands support in markdown. See https://github.com/microsoft/vscode-languageserver-node/blob/main/client/src/common/client.ts#L375

**Comment by JustinGrote** [comment]
Ah I see it's a language client option, not a specific message flag. That makes sense.
